src/domain/Lists/DraggableList/DraggableList.tsx

```diff
@@ -0,0 +1,107 @@
+import React from 'react'
+import {
+  DragDropContext,
+  Draggable,
+  DropResult,
+  Droppable
+} from 'react-beautiful-dnd'
+import styled, { css } from 'styled-components'
+
+import { Props, Variables } from '../../../common'
+
+interface IDraggableListProps<T> {
+  /** The array of items */
+  items: T[]
+  /** Function to update the array of items when they have been re-arranged */
+  setItems: (nodes: T[]) => void
+  /** An array of ReactNode that should represent the array of items */
+  children: React.ReactNode[]
+  /** The data-component-context */
+  componentContext?: string
+}
+
+interface IListItemProps {
+  isDragging: boolean
+  children: React.ReactNode
+}
+
+const StyledListItem = styled.div<IListItemProps>`
+  border: 1px solid ${Variables.Color.n250};
+  background-color: ${Variables.Color.n100};
+  padding: ${Variables.Spacing.sMedium}px;
+  border-radius: ${Variables.Style.borderRadius}px;
+  box-shadow: ${Variables.BoxShadow.bsLv1Static};
+
+  ${(props) => props.isDragging && css`
+    box-shadow: ${Variables.BoxShadow.bsLv3Active};
+    background-color: ${Variables.Color.n150};
+    cursor: grabbing !important;
+  `}
+`
+
+const StyledListArea = styled.div`
+  display: grid;
+  row-gap: ${Variables.Spacing.sXSmall}px;
+`
+
+function DraggableList<T> ({ items, setItems, children, componentContext }: IDraggableListProps<T>) {
```

Comment:
not sure if you want to use arrow function

`const DraggableList = <T extends {}>({ items, setItems, children, componentContext }: IDraggableListProps <T>) => {
....
}`